### DISCUSSION TITLE
Issues with "app_type = ThermalHydraulicsApp"

### DISCUSSION THREAD

**Discussion by amerallaf** [idaholab/moose#27329] [Q&A Modules: General] [answered]
Hello,
I am new to MOOSE and I am trying to learn from the example (link below) CoupledHeatTransferAction in coupling to develope a model that couples heat transfer and thermal hydraulics. 
https://mooseframework.inl.gov/source/actions/CoupledHeatTransferAction.html
However, I am having the following error: 

*** ERROR ***
"ThermalHydraulicsApp" is not a registered application name.
No search paths were set. We made no attempts to locate the corresponding library file.
Make sure you have compiled the library and either set the "library_path" variable in your input file or exported "MOOSE_LIBRARY_PATH".

I have both the thermal_hydraulics and the heat_transfer packages active in the makefile, and both work fine in other practices.
I also tried to dynamically register the object, by the following:
[Problem]
  register_objects_from = 'ThermalHydraulicsApp'
  library_path = '../thermal_hydraulics/lib'
[]
and still the outcome is the same.

The only time this inputs run, is when I remove "app_type = ThermalHydraulicsApp", but then I am concerned if the heat transfer is captured properly.

What do you recommend me to do to overcome this issue?

**Comment by GiudGiud** [comment, marked as answer]
Hello

which executable are you using ?

specifying the app type has no incidence on the results. It s a micro optimization 

Guillaume

**Reply by amerallaf** [reply]
Oh Thank you so much for answering, so it is not big deal if it is not working.
Sorry what do you mean by "excutable"? I am using Linux ubuntu (terminal in windows), and activate MOOSE through Conda, does this help answering your question?

**Reply by GiudGiud** [reply]
The executable is the -opt file you use to run your inputs

On windows they are .exe files

**Reply by amerallaf** [reply]
Thank you for the clarification, yes it is -opt excutable.
